### PR TITLE
Use File#isFile rather then File#exists across the codebase

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: src/test/java/com/virtuslab/archunit/ForbiddenMethodsTestSuite.java [hunk 1]
```diff
@@ -32,6 +32,14 @@ public void no_classes_should_call_Collections_nCopies() {
         .check(importedClasses);
   }
 
+  @Test
+  public void no_classes_should_call_File_exists() {
+    noClasses()
+        .should().callMethod(java.io.File.class, "exists")
+        .because("in most cases, the check you want to do is `isFile` rather than `exists` (what if this is a directory?)")
```
Comment (PawelLipski) [inline src/test/java/com/virtuslab/archunit/ForbiddenMethodsTestSuite.java:39]: I literally don't remember a case when I wanted to know whether this is a file OR directory, ever. And this is exactly what `exists` does xD

File: backendImpl/src/test/resources/reference-cli-version.properties [hunk 2]
```diff
@@ -1 +1 @@
-referenceCliVersions=2.15.9,2.16.1
+referenceCliVersions=2.15.9,2.16.1,3.0.0
```
Comment (PawelLipski) [inline backendImpl/src/test/resources/reference-cli-version.properties:1]: Meh... this should be done with https://www.testcontainers.org/ one day :/

Comment (PawelLipski) [reply backendImpl/src/test/resources/reference-cli-version.properties:1]: https://github.com/VirtusLab/git-machete-intellij-plugin/issues/703

**Review by mkondratek — APPROVED** [review]
lgtm

**Pull request merged**